js/ClockCtrl.js

```diff
@@ -1,348 +1,269 @@
-        return map;
-    }, {});
-}
-
-angular.module('Clock',['ngStorage'])
-    .filter('time',function() {
-        function pad(str) {
-            str = ''+str;
-            if (str.length<2) {
-                str = '0'+str;
-            }
-            return str;
-        }
-        return function(msec,hundreds) {
-            var secs = Math.floor(msec/1000);
-            var hsec = Math.floor((msec/100)) % 10;
-            var sec = secs % 60;
-            var min = Math.floor((secs-sec)/60);
-            var str = min+':'+pad(sec);
-            if (hundreds) {
-                str += '.'+hsec;
-            }
-            return str;
-        };
-    })
-    .factory('$audio',[
-        function() {
-            function init(file,cb) {
-                new Audio5js({
-                    swf_path: 'swf/audio5js.swf',
-                    ready: function () {
-                        this.load(file);
-                        this.reset = function() {
-                            this.pause();
-                            this.load(file);
-                        };
-                        cb(this);
-                    }
-                });
-            }
-
-            return {
-                init: init
-            };
-        }
-    ])
-
-    .controller('ClockCtrl',[
-        '$scope','$timeout','$audio','$window','$localStorage',
-        function($scope,$timeout,$audio,$window,$localStorage) {
-            //initial values
-            $audio.init('mp3/lossetrack-A +6.mp3',function(track) {
-                $scope.runTrack = track;
-            });
-            $audio.init('mp3/lossetrack-B.mp3',function(track) {
-                $scope.stopTrack = track;
-            });
-            //initialize config with the angular configuration
-            var urlConfig
-            try {
-                // console.log(params().state);
-                urlConfig = JSON.parse(params().state);
-            } catch(e) {
-                //no url Config
-            }
-
-            //config from localStorage, then url, then defaults from config
-            $scope.$storage = $localStorage.$default({
-                config: urlConfig || clockConfig
-            });
-            var handlers = {};
-            $scope.bgColor = 'black';
-            $scope.state = 'stopped';
-            $scope.time = $scope.$storage.config.seconds * 1000;
-            $scope.armTime = $scope.$storage.config.seconds * 1;
-            $scope.tenths = false;
-            $scope.size = 340;
-
-            $scope.pos = {
-                x: 0,
-                y: 0
-            };
+angular.module('Clock').controller('ClockCtrl',[
+    '$scope','$timeout','$window','$config','$tracks',
+    function($scope,$timeout,$window,$config,$tracks) {
+        $scope.config = $sconfig.get();
+        $scope.time = $scope.config.seconds * 1000;
+        $scope.armTime = $scope.config.seconds * 1;
+        $scope.connect();

```

Comment:
This cannot be called here as it is not defined yet

Reply:
Moved it